stock_orderpoint_automatic_creation/models/product.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# (c) 2016 Daniel Campos <danielcampos@avanzosc.es> - Avanzosc S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, api
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.model
+    def create(self, vals):
+        result = super(ProductProduct, self).create(vals)
+        orderpoint_obj = self.env['stock.warehouse.orderpoint']
+        wh_obj = self.env['stock.warehouse']
+        if result.type != 'service':
+            if self.env.user.company_id.create_orderpoints:
+                company = self.env.user.company_id
+                warehouses = wh_obj.search([('company_id', '=', company.id)])
+                for warehouse in warehouses:
```

Comment:
What do you think the next idea?
Add a selection field product category and company with Yes, No or None values to determine whether or not created the rule.
If this field is set to None in the product, the value is obtained from the search. If the category also has None is obtained from the company.
This selection field should be company_dependent.